app/models/manageiq/providers/openstack/cloud_manager/vm/resize.rb

```diff
@@ -1,9 +1,18 @@
 module ManageIQ::Providers::Openstack::CloudManager::Vm::Resize
-  def validate_resize
-    msg = validate_vm_control
-    return {:available => msg[0], :message => msg[1]} unless msg.nil?
-    return {:available => true, :message => nil} if %w(ACTIVE SHUTOFF).include?(raw_power_state)
-    {:available => false, :message => _("The Instance cannot be resized, current state has to be active or shutoff.")}
+  extend ActiveSupport::Concern
+
+  included do
+    supports :resize do
+      msg = validate_vm_control
```

Comment:
This is exactly the reason why we want to get rid of things like `validate_vm_control`. Having a ternary method is bad, but a method that has a ternary return value is even worse.

> supports_control will either return true/false, eliminating the flexibility "nil" provided in original validate_vm_control method.

In our case `supports_control? # => true` means, continue checking the additional conditions. Which is basically the nil case.

Have a look at [this](https://github.com/ManageIQ/manageiq/pull/9271/files#diff-40a469981f9f59cdb22f44b221fc3837) change. If `validate_vm_control` returned nil, then the operation was supported.

So for resize this would be
```ruby
supports :resize do
  unsupported_reason_add(:resize, unsupported_reason(:control)) unless supports_control?
  unsupported_reason_add(:resize,  _("The Instance cannot be resized, current state has to be active or shutoff.")) unless raw_power_state.in? %w(ACTIVE SHUTOFF)
end
```

It would also be great if you could also add tests, that check the return value of `supports_resize?`